### ISSUE TITLE
TUI: can't see string arguments longer than 127 characters

### ISSUE THREAD

**Issue by shykes** [dagger/dagger#9212]
## Overview

There seems to be a built-in protection to avoid printing string arguments that are too long. But this is not practical when using modules that involve passing long shell commands as arguments. Specifically, when using [daggy](https://daggerverse.dev/mod/github.com/shykes/x/daggy), which involves a LLM running dagger shell commands, and tracing is the only practical way to know what commands the LLM is running, this really hurts the experience.

I don't know what the best solution: just lift the limit to a higher number? Remove the limit altogether? Truncate to the first N characters rather than showing a digest? To be discussed.

More detailed repro below.

## What I did

Call a function with a string argument that is longer than 127 characters.

For example:

```console
dagger shell -c 'github.com/shykes/hello | hello --name aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa'
```

## What I expected

The TUI shows the full argument value in the trace:

```console
✔ connect 0.2s
✔ load module 2.3s
✔ load module 1.7s

✔ hello: Hello! 0.0s
✔ .hello(name: "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"): String! 0.1s
```

## What I saw

Instead of the argument value, the TUI shows a digest:

```console
✔ connect 0.2s
✔ load module 1.5s
✔ load module 1.7s

✔ hello: Hello! 0.0s
✔ .hello(name: ETOOBIG:sha256:6836cf13bac400e9105071cd6af47084dfacad4e5e302c94bfed24e013afb73e): String! 0.1s
```

**Comment by cwlbraa** [comment]
sometimes the truncation feels valuable, no? like if you've got CA certs or binary content floating around, printing them creates serious ui gore. perhaps it's sensible to configure this for specific functions or even calls? no idea what the interface would look like for either.

**Comment by shykes** [comment]
> sometimes the truncation feels valuable, no? like if you've got CA certs or binary content floating around, printing them creates serious ui gore. perhaps it's sensible to configure this for specific functions or even calls? no idea what the interface would look like for either.

Yeah a balance is needed, I agree. How about:

1. Full string is always sent in telemetry (not sure if that's the case today). So what to render is purely a frontend decision, and can be changed over time
2. Default rendering is to truncate at a) first newline, or c) first non-printable character, or c) 1000 characters, whichever comes first
3. When rendering a truncated argument: a) the first line/first 1000 lines/first printable characters are directly visible, b) there is a cue that truncating has taken place, c) the UI has an affordance to expand the line
4. In TUI, the affordance to expand would be pressing enter and/or increasing verbosity. In web, it would be clicking the "more" or "expand" link?

In any case, showing a digest by default is never useful IMO

**Comment by vito** [comment]
Yeah this was a game-time decision. The data is all there, the UI just truncates. This was originally added to avoid showing the absolutely ginormous schema introspection JSON that gets passed to modules.

The heuristics mentioned above sound reasonable.

The use case for showing a digest is so you can at least tell when the gigantic string is different from previous/other calls, which is hard to tell with truncation, or even with eyeballing the full value if it's large. This came in handy in some scenarios but maybe it's too fringe and non-obvious.
